main_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2019 Tim Waugh
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import "testing"
+
+func TestGetTemplate(t *testing.T) {
+	tcs := []struct {
+		name     string
+		args     []string
+		expected string
+	}{
+		{
+			"default",
+			[]string{"retrodep", "."},
+			defaultTemplate,
+		},
+		{
+			"go-template",
+			[]string{"retrodep", "-o", "go-template={{.Pkg}}", "."},
+			"{{.Pkg}}",
+		},
+		{
+			"compatibility",
+			[]string{"retrodep", "-template", "@{{.Rev}}", "."},
+			"{{.Pkg}}@{{.Rev}}",
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
```

Comment:
What effect does this re-assignment have?

Reply:
The `tc` used by the for loop gets a different value each time round the loop. The `t.Run` call runs the provided function in its own goroutine, so `tc` in that function comes from the closure for running it in. With `tc := tc` we create a new variable for each new goroutine we create, ensuring they don't all run for the same `tc`.

More information: https://golang.org/doc/faq#closures_and_goroutines

Reply:
Can you add a comment to this code? Unaware people might remove this. :grimacing:

Reply:
Got it. Thanks @twaugh